Tests/AfluentTests/SubscriptionTests.swift

```diff
@@ -159,4 +160,376 @@ final class SubscriptionTests: XCTestCase {
+        actor Test {
+            var started = false
+            var ended = false
+
+            func start() { started = true }
+            func end() { ended = true }
+        }
+        let test = Test()
+
+        let exp = expectation(description: "thing happened")
+        exp.isInverted = true
+        let sequence = AsyncStream<Void> { continuation in
+            Task {
+                await test.start()
+                try await Task.sleep(for: .milliseconds(10))
+                continuation.yield()
+            }
+        }.map {
+            await test.end()
+            exp.fulfill()
+        }
+
+        sequence.subscribe()
+            .store(in: &set)
+
+        try await Task.sleep(for: .milliseconds(2))
+
+        set.removeAll()
+
+        await fulfillment(of: [exp], timeout: 0.011)
+
+        let started = await test.started
+        let ended = await test.ended
+
+        XCTAssert(started)
+        XCTAssertFalse(ended)
+    }
+
+    func testAsyncSequenceCancelledViaDeinitialization_WhenStoredInCollection() async throws {
+        try XCTSkipIf(ProcessInfo.processInfo.environment["CI"] == "true")
+        actor Test {
+            var started = false
+            var ended = false
+
+            func start() { started = true }
+            func end() { ended = true }
+        }
+        let test = Test()
+
+        let exp = expectation(description: "thing happened")
+        exp.isInverted = true
+        let sequence = AsyncStream<Void> { continuation in
+            Task {
+                await test.start()
+                try await Task.sleep(for: .milliseconds(10))
+                continuation.yield()
+            }
+        }.map {
+            await test.end()
+            exp.fulfill()
+        }
+
+        sequence.subscribe()
+            .store(in: &collection)
+
+        try await Task.sleep(for: .milliseconds(2))
+
+        collection.removeAll()
+
+        await fulfillment(of: [exp], timeout: 0.011)
+
+        let started = await test.started
+        let ended = await test.ended
+
+        XCTAssert(started)
+        XCTAssertFalse(ended)
+    }
+
+    func testAsyncSequenceReceivesCompletionWhenCancelled() async throws {
+        try XCTSkipIf(ProcessInfo.processInfo.environment["CI"] == "true")
+        actor Test {
+            var started = false
+            var ended = false
+
+            func start() { started = true }
+            func end() { ended = true }
+        }
+        let test = Test()
+
+        let exp = expectation(description: "thing happened")
+        exp.isInverted = true
+        let sequence = AsyncThrowingStream<Void, Error> { continuation in
+            Task {
+                await test.start()
+                try await Task.sleep(for: .milliseconds(10))
+                continuation.yield()
+            }
+        }.map {
+            await test.end()
+            exp.fulfill()
+        }
+
+        let completedSubject = SingleValueSubject<Void>()
```

Comment:
Super nit picky, prefer `SingleValueChannel` when you can afford to `await channel.send()`